modules/opengl/glsl/utils/selectioncolor.glsl

```diff
@@ -0,0 +1,50 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2022 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ * 
+ *********************************************************************************/
+
+#ifndef IVW_SELECTIONCOLOR_GLSL
+#define IVW_SELECTIONCOLOR_GLSL
+
+struct SelectionColor {
+    vec4 color;
+    float colorMixIn;
+    float alphaMixIn;
+    bool visible;
+};
+
+/**
+ * Blend @p srcColor with @p brushingColor using separate interpolants for color and alpha.
+ * @param srcColor
+ * @param brushingColor
+ * @return color resulting from mixing @p srcColor with @p brushingColor
+ */
+vec4 applySelectionColor(in vec4 srcColor, in SelectionColor brushingColor) {
+    return mix(srcColor, brushingColor.color, vec4(vec3(brushingColor.colorMixIn), brushingColor.alphaMixIn));
```

Comment:
The visible state is never used. There should be a check here?

Reply:
I see. The code becomes a bit diluted though as the struct is defined just above and then the bool unused.

Reply:
With respect to the helper function, yes. But not regarding the SelectionColorProperty. I also rather have a `discard()` directly in the shader instead of some utility function.